### PR TITLE
OSDOCS-11772: Upgrade ROSA with HCP cluster through  Openshift Cluster manager console

### PR DESCRIPTION
**Pull request by fmcdonal**
Version(s):
4.17

Issue:
https://issues.redhat.com/browse/OSDOCS-11772

Link to docs preview:
https://85588--ocpdocs-pr.netlify.app/
https://85588--ocpdocs-pr.netlify.app/openshift-rosa-hcp/latest/upgrading/rosa-hcp-upgrading.html
https://85588--ocpdocs-pr.netlify.app/openshift-rosa/latest/upgrading/rosa-hcp-upgrading.html
https://85588--ocpdocs-pr.netlify.app/openshift-rosa/latest/upgrading/rosa-upgrading-sts.html

New note in:
https://85588--ocpdocs-pr.netlify.app/openshift-rosa/latest/upgrading/rosa-hcp-upgrading#rosa-hcp-upgrading-cli-machinepool_rosa-hcp-upgrading-whole-cluster

https://85588--ocpdocs-pr.netlify.app/openshift-rosa-hcp/latest/upgrading/rosa-hcp-upgrading#rosa-hcp-upgrading-cli-machinepool_rosa-hcp-upgrading-whole-cluster

QE review:
- [ ] QE none needed

### CODE REVIEW

File: modules/rosa-upgrading-manual-ocm.adoc [hunk 1]
```diff
@@ -3,6 +3,12 @@
 // * rosa_upgrading/rosa-upgrading.adoc
 // * rosa_upgrading/rosa-upgrading-sts.adoc
 
+ifdef::openshift-rosa[]
+ifeval::["{context}" == "rosa-hcp-upgrading"]
+:hcp-in-rosa:
+endif::[]
+endif::openshift-rosa[]
```
Comment (adellape) [inline modules/rosa-upgrading-manual-ocm.adoc:10]: Added for the specific scenario where there is an HCP-only procedure that is appearing in the `openshift-rosa` distro.

I.e., the `rosa-hcp-upgrading` assembly appears in the `openshift-rosa` distro in this case. This temporary `hcp-in-rosa` entity allows using `ifdef`/`ifndef` to target that situation.

**Comment by fmcdonal** [comment]
@kdoberst and @bchandra-ocp and @andreadecorte please review for correctness, thanks.

**Comment by bchandra-ocp** [comment]
1. Following section refers to CLI fields but are mentioned in the overview section. Further, it only mentions two configurations (max unavailable and max surge) but there is also configuration like max node drain timeout that are relevant to upgrades that must be mentioned.
image

2. Upgrading HCP cluster as a whole appears to be repeating the same content under upgrading HCP and upgrading machine pools respectively. The whole section about upgrading cluster as a whole is not required. Perhaps leave a statement that ROSA HCP requires upgrading HCP and machine pools separately (it's already in the section's introduction)
Further, upgrading with CLI is repeated as well. 
image

4. If I am right, Console only supports updating HCP - not updating machine pools.

**Comment by kdoberst** [comment]
Correct in the OCMUI the HCP machine pools need to be upgraded manually separately from the control plane.    You can set a schedule for the control plane but, for now in the UI, the user needs to manually update the machine pools.   The option to upgrade the machine pools only happens when the control plane is at the latest version.  The only option the user has is to upgrade the machine pool(s) to the same version as the control plane.

I'll get screenshots in a little bit - currently our staging environment is down.

**Comment by bchandra-ocp** [comment]
Also note, in the UI while it's possible to delete a scheduled upgrade of HCP, it's not yet possible to delete the scheduled upgrade of machine pool.

**Comment by kdoberst** [comment]
I think I caused  some confusion - the content as written is correct.  I was trying to add additional information about how updating the machine pools is different in the UI for HCP clusters.   Up to you all if adding additional information about updating machine pools in HCP is needed.  Personally, I don't think so at this point, but let's get all the info out while I'm here :-) 

To summarize:
In ROSA Classic - when you update in the UI - both the control plane and machine pools are updated to the same version at the same time.

In ROSA Hosted - when you update in the UI - you are only updating the control plane.  The machine pools will stay at the same version unless the user uses the CLI or comes back to the UI and  updates the machine pools manually.  The user is only given the option to update the machine pools when the control plane version is at the latest version available.  At that point, the user only has the option to update the machine pools to the latest version immediately.    The user would need to use the CLI, for now, if they want to specify the exact version or schedule the update of a machine poo(s).

I created a video of 2 HCP clusters.  The first is at a lower version and it shows the process of scheduling an update for the control plane.  It shows that there is no option to update the node pools.  The second cluster is where the control plan is at the latest version, but the machine pools are not.  It shows that option to update the machine pool to the latest version.

https://drive.google.com/drive/folders/1d9W7WlGD8Iyh8K3YcpuXf607XmKgty_j

I am sorry for the slow response - our environment was down for most of the day yesterday.

**Comment by fmcdonal** [comment]
@bchandra-ocp there seems to be confusion here. Km has reviewed and current content is correct. 
The card is for : https://issues.redhat.com/browse/OSDOCS-11772
Upgrading ROSA with HCP clusters through the Openshift Cluster manager console.
So if we need to revisit steps for CLI it will need to be in a new card.
If there is an issue with content can you possibly highlight the content and then add the comment to the doc/section of the doc we need to change and put before and after suggestions so we can move on quickly to get this closed before PTO starts.
If this is ok for the console, as it stands can we go ahead with this peer and QE review and then if you can create another card for CLI repeats we can do that, but scope of this is about the console and Kim can advise on that but not on CLI, if that makes sense, thanks

**Comment by kdoberst** [comment]
From the UI/console perspective  - LGTM

**Comment by fmcdonal** [comment]
@bchandra-ocp we have generated a new card for CLI updates and are going to implement these changes for console first as the console team has worked on this card and PR. thanks

**Comment by fmcdonal** [comment]
SME has already peer reviewed so no QE review needed

**Pull request merged**